### PR TITLE
Ceph REST API: Compatibility fixes for Python 3

### PR DESCRIPTION
**Pull request by onyb**
- Includes a simple Jinja template for displaying the API docs.
- The `ceph_rest_api` module goes to a separate folder under `pybind`

**Screenshot**
2016-07-20-161424_1366x768_scrot

### CODE REVIEW

File: src/pybind/ceph-rest-api/static/css/main.css [hunk 1]
```diff
@@ -131,11 +131,14 @@
...
+    height: auto;
+    overflow-y: auto;
     padding-top: 10px;
+    padding-bottom: 10px;
     padding-left: 12px;
     border-radius: 7px;
-    font-family: "Courier New"
+    font-family: "Monospace";
+    font-size: 12px;
 }
```
Comment (jdurgin) [inline src/pybind/ceph-rest-api/static/css/main.css]: can squash this into the commit that added this file

**Comment by jdurgin** [comment]
@dmick would you like to look at this as well? the py3 fixes look good to me

File: src/pybind/ceph-rest-api/static/css/main.css (hunk 1 above)
Comment (onyb) [inline src/pybind/ceph-rest-api/static/css/main.css]: @jdurgin Done.

**Comment by dmick** [comment]
I haven't reviewed this super-carefully, but it looks like most of this is not really about python 3, but rather some welcome restructuring for template-driven rendering?...

File: src/Makefile-server.am [hunk 2]
```diff
@@ -6,7 +6,7 @@ bin_SCRIPTS += \
 	ceph-debugpack \
 	ceph-crush-location
 
-python_PYTHON += pybind/ceph_rest_api.py
+python_PYTHON += pybind/ceph-rest-api
```
Comment (jdurgin) [inline src/Makefile-server.am]: need to update CMakeLists.txt for this too

**Comment by jdurgin** [comment]
@onyb could you provide a bit more context for your api docs changes, perhaps with a link to a screenshot?

**Comment by onyb** [comment]
@dmick and @jdurgin I've uploaded the screenshot. Please check it out.

I launched the API server after a CMake build (with Python `rados` package installed and `LD_LIBRARY_PATH`, `PYTHONPATH` set) using the command: `./src/ceph-rest-api --id admin`

**Comment by onyb** [comment]
@jdurgin After discussion with @tchaikov, I updated `CMakeLists.txt` and organized the code inside the package for a more natural import. I don't think the build failure is related, since I did a `make install` and `ceph-rest-api` worked fine after that. If this PR looks good, could you please add it to a branch in ceph.git so that I can test my workunit patches with it?

**Comment by dmick** [comment]
Yes, the build failures were not yours; if you rebase, they should disappear.

**Comment by jdurgin** [comment]
lgtm

**Comment by jdurgin** [comment]
The distro packaging needs fixing for the new directory scheme (ceph.spec.in and debian/ceph-mon.install):

http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-deb-trusty-amd64-basic/log.cgi?log=8ae8fd0131053a2a2dde0670c4fb319acdb7f284
http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-rpm-centos7-amd64-notcmalloc/log.cgi?log=8ae8fd0131053a2a2dde0670c4fb319acdb7f284

**Comment by onyb** [comment]
@jdurgin I pushed a commit for the distro packages. Could you re-test it please?

**Comment by dachary** [comment]
needs rebase

**Comment by liewegas** [comment]
this needs rebase.  feel free to reopen if this is still relevant.  we probably want to focus api efforts on the one in ceph-mgr, though!

**Pull request closed without merging**